lib/trackler/problem.rb

```diff
@@ -7,81 +7,108 @@ class Problem
     def initialize(slug, root)
       @slug = slug
       @root = root
+      @xcommon_repository_url = 'https://github.com/exercism/x-common/blob/master'
     end
 
     def exists?
-      !!md && !!yaml
+      description_exists? && yaml_exists?
     end
 
     def name
-      @name ||= slug.split('-').map(&:capitalize).join(' ')
+      @name ||= slug_to_name
     end
 
     def description
-      @description ||= File.read(filepath(md))
+      @description ||= load_description
```

Comment:
This one was right on the border for me, I ended up extracting it to be consistent with the level of abstraction of `name` and to group all the methods that depend on `File` together.